test/model/actions_test.dart

```diff
@@ -121,6 +145,30 @@ void main() {
       async.elapse(unregisterDelay - TestGlobalStore.removeAccountDuration);
       check(newConnection.isOpen).isFalse();
     }));
+
+    test('notifications are removed after logout', () => awaitFakeAsync((async) async {
+      await prepare();
+      testBinding.firebaseMessagingInitialToken = '123';
+      addTearDown(NotificationService.debugReset);
+      NotificationService.debugBackgroundIsolateIsLive = false;
+      await runWithHttpClient(NotificationService.instance.start);
+
+      // Create a notification to check that it's removed after logout
+      final message = eg.dmMessage(from: eg.otherUser, to: [eg.selfUser]);
+      testBinding.firebaseMessaging.onMessage.add(
+        RemoteMessage(data: messageFcmMessage(message).toJson()));
+      async.flushMicrotasks();
+      check(testBinding.androidNotificationHost.activeNotifications).isNotEmpty();
+
+      check(testBinding.globalStore).accountIds.single.equals(eg.selfAccount.id);
+      final future = logOutAccount(testBinding.globalStore, eg.selfAccount.id);
+      check(testBinding.globalStore.takeDoRemoveAccountCalls())
+        .single.equals(eg.selfAccount.id);
+      async.elapse(TestGlobalStore.removeAccountDuration);
+      await future;
+
+      check(testBinding.androidNotificationHost.activeNotifications).isEmpty();
```

Comment:
This can be further simplified, on the lines described in https://github.com/zulip/zulip-flutter/pull/1349#discussion_r2021897865 :
```suggestion
      check(testBinding.androidNotificationHost.activeNotifications).isNotEmpty();

      await logOutAccount(testBinding.globalStore, eg.selfAccount.id);
      check(testBinding.androidNotificationHost.activeNotifications).isEmpty();
```

(with also a tweak to join checks up to the steps they're checking, https://github.com/zulip/zulip-flutter/pull/1349#discussion_r2021889137)

Reply:
This way the narrative is: look, there were notifications; then we called logout; now there aren't any, so whatever was there got removed.